src/vs/code/electron-main/app.ts

```diff
@@ -336,7 +336,11 @@ export class CodeApplication extends Disposable {
 
 		// We handle uncaught exceptions here to prevent electron from opening a dialog to the user
 		setUnexpectedErrorHandler(error => this.onUnexpectedError(error));
-		process.on('uncaughtException', error => onUnexpectedError(error));
+		process.on('uncaughtException', error => {
+			if (!isSigPipeError(error)) {
```

Comment:
Should we move this into the `onUnexpectedError`?

Reply:
I thought about that initially when doing the first pass of this, but opted not to for two reasons:

- it's called from many more places where the loop only happens in `process.on('uncaughtException`
- we don't know if `errorHandler.onUnexpectedError` will actually log to the console and cause a loop

...but that may be a better idea considering the virality of this check. What do you think?

Reply:
Ah ok, I think its fine then.